zebra-network/src/protocol/external/tests/vectors.rs

```diff
@@ -24,3 +35,254 @@ fn parses_msg_wtx_inventory_type() {
 
     assert_eq!(deserialized, InventoryHash::Wtx([0u8; 64].into()));
 }
+
+/// Test that deserializing [`AddrV1`] into [`MetaAddr`] succeeds,
+/// and produces the expected number of addresses.
+///
+/// Also checks some of the deserialized address values.
+#[test]
+fn parses_msg_addr_v1_ip() {
+    zebra_test::init();
+
+    let codec = Codec::builder().finish();
+
+    for (case_idx, addr_v1_bytes) in zebra_test::network_addr::ADDR_V1_IP_VECTORS
+        .iter()
+        .enumerate()
+    {
+        let mut addr_v1_bytes = io::Cursor::new(addr_v1_bytes);
+
+        let deserialized: Message = codec
+            .read_addr(&mut addr_v1_bytes)
```

Comment:
Minor: I think you can avoid the `Cursor` if you want to:

```suggestion
        let deserialized: Message = codec
            .read_addr(&mut addr_v1_bytes.as_slice())
```

Reply:
Fixed in 96e6121aa